test/integration/consul-container/libs/cluster/cluster.go

```diff
@@ -121,6 +123,60 @@ func (c *Cluster) Remove(n libagent.Agent) error {
 	return nil
 }
 
+// StandardUpgrade upgrades a running consul cluster following the steps from
+//
+//	https://developer.hashicorp.com/consul/docs/upgrading#standard-upgrades
+//
+// - takes a snapshot
+// - terminate and rejoin the new version of consul
+func (c *Cluster) StandardUpgrade(t *testing.T, ctx context.Context, targetVersion string) error {
+	execCode, err := c.Agents[0].Exec(context.Background(), []string{"consul", "snapshot", "save", "backup.snap"})
+	if execCode != 0 {
+		return fmt.Errorf("error taking snapshot of the cluster, returned code %d", execCode)
+	}
+	if err != nil {
+		return err
+	}
+
+	// verify only the leader can take a snapshot
+	snapshotCount := 0
+	for _, agent := range c.Agents {
+		files, err := ioutil.ReadDir(filepath.Join(agent.DataDir(), "raft", "snapshots"))
+		if err != nil {
+			return err
+		}
+		if len(files) >= 1 {
+			snapshotCount++
+		}
+	}
+
+	if snapshotCount != 1 {
```

Comment:
Can a leader have more than 1 snapshot file? I assumed `snapshotCount` is the total number of snapshots found in the directory

Reply:
Yes, consul will take snapshot periodically or on the snapshot command. But since the test takes snapshot in a short amount of time after agent is started, there must be only one snapshot file.